scheduler/networktopology/network_topology.go

```diff
@@ -56,5 +101,232 @@ func NewNetworkTopology(cfg *config.Config, rdb redis.UniversalClient, resource
 
 // Peek returns the oldest probe without removing it.
 func (n *networkTopology) Peek(src, dest string) (*Probe, bool) {
-	return nil, false
+	key := fmt.Sprintf("probes:%s:%s", src, dest)
+	jsonStr, err := n.rdb.LIndex(context.Background(), key, 0).Result()
```

Comment:
What is `jsonStr`?

Reply:
Because Probe stores in redis as a string parsed into json, called jsonStr.